tools/install-powershell.sh

```diff
@@ -1,154 +1,155 @@
+                PSUEDONAME=`cat /etc/redhat-release | sed s/.*\(// | sed s/\)//`
+                REV=`cat /etc/redhat-release | sed s/.*release\ // | sed s/\ .*//`
+            elif [ -f /etc/system-release ] ; then
+                DIST=`cat /etc/system-release |sed s/\ release.*//`
+                PSUEDONAME=`cat /etc/system-release | sed s/.*\(// | sed s/\)//`
+                REV=`cat /etc/system-release | sed s/.*release\ // | sed s/\ .*//`
+                if [[ $DIST == *"Amazon Linux"* ]] ; then
+                    DistroBasedOn='amazonlinux'
+                else
+                    DistroBasedOn='redhat'
+                fi
+            elif [ -f /etc/SuSE-release ] ; then
+                DistroBasedOn='suse'
+                PSUEDONAME=`cat /etc/SuSE-release | tr "\n" ' '| sed s/VERSION.*//`
+                REV=`cat /etc/SuSE-release | grep 'VERSION' | sed s/.*=\ //`
+            elif [ -f /etc/mandrake-release ] ; then
+                DistroBasedOn='mandrake'
+                PSUEDONAME=`cat /etc/mandrake-release | sed s/.*\(// | sed s/\)//`
+                REV=`cat /etc/mandrake-release | sed s/.*release\ // | sed s/\ .*//`
+            elif [ -f /etc/debian_version ] ; then
+                DistroBasedOn='debian'
+                DIST=`cat /etc/lsb-release | grep '^DISTRIB_ID' | awk -F=  '{ print $2 }'`
+                PSUEDONAME=`cat /etc/lsb-release | grep '^DISTRIB_CODENAME' | awk -F=  '{ print $2 }'`
+                REV=`cat /etc/lsb-release | grep '^DISTRIB_RELEASE' | awk -F=  '{ print $2 }'`
             fi
-        elif [ -f /etc/SuSE-release ] ; then
-            DistroBasedOn='suse'
-            PSUEDONAME=`cat /etc/SuSE-release | tr "\n" ' '| sed s/VERSION.*//`
-            REV=`cat /etc/SuSE-release | grep 'VERSION' | sed s/.*=\ //`
-        elif [ -f /etc/mandrake-release ] ; then
-            DistroBasedOn='mandrake'
-            PSUEDONAME=`cat /etc/mandrake-release | sed s/.*\(// | sed s/\)//`
-            REV=`cat /etc/mandrake-release | sed s/.*release\ // | sed s/\ .*//`
-        elif [ -f /etc/debian_version ] ; then
-            DistroBasedOn='debian'
-            DIST=`cat /etc/lsb-release | grep '^DISTRIB_ID' | awk -F=  '{ print $2 }'`
-            PSUEDONAME=`cat /etc/lsb-release | grep '^DISTRIB_CODENAME' | awk -F=  '{ print $2 }'`
-            REV=`cat /etc/lsb-release | grep '^DISTRIB_RELEASE' | awk -F=  '{ print $2 }'`
-        fi
-        if [ -f /etc/UnitedLinux-release ] ; then
-            DIST="${DIST}[`cat /etc/UnitedLinux-release | tr "\n" ' ' | sed s/VERSION.*//`]"
+            if [ -f /etc/UnitedLinux-release ] ; then
+                DIST="${DIST}[`cat /etc/UnitedLinux-release | tr "\n" ' ' | sed s/VERSION.*//`]"
+            fi
+            OS=`lowercase $OS`
+            DistroBasedOn=`lowercase $DistroBasedOn`
+            local DIST
+            local DistroBasedOn
+            local PSUEDONAME
+            local REV
```

Comment:
Shall we move the `local` variable declarations up beside `OS`, `Kernel` and `Mach`?